server/src/main/java/org/elasticsearch/action/admin/cluster/configuration/AddVotingConfigExclusionsRequest.java

```diff
@@ -47,7 +53,7 @@
      * @param nodeDescriptions Descriptions of the nodes to add - see {@link DiscoveryNodes#resolveNodes(String...)}
      */
     public AddVotingConfigExclusionsRequest(String[] nodeDescriptions) {
```

Comment:
If we migrate these tests now to use node names instead of descriptions, I'm a bit concerned that we may not have tests to prove that the changes are still backward compatible and don't have bugs that may break logic based on nodeDescriptions, before it is fully migrated to nodeIds / nodeNames. What do you think?

Reply:
Sorry, you're right, I didn't quite mean "all" these tests. We should comprehensively test the different kinds of node resolution by strengthening `AddVotingConfigExclusionsRequestTests`. The other tests can move over to node names without loss of coverage IMO.

Reply:
Make sense.

Reply:
Sorry missed this earlier. Done in commit https://github.com/elastic/elasticsearch/pull/50836/commits/02a353317d415560ba94947771f09e7fa1257277

Reply:
You removed this useful constructor in 02a3533 and added a lot of noise to the tests as a result. Could you follow my suggestion [above](https://github.com/elastic/elasticsearch/pull/50836#discussion_r386254430) instead?

Reply:
Sorry when I re-read this thread I somehow got the wrong idea that this constructor need to be removed. I reverted that commit and tried again in commit https://github.com/elastic/elasticsearch/pull/50836/commits/5c7a226f29540e46f5d6e26db8722e1f8123db69. Could you let me know if this looks good?